app/inventory/domain/domain.ts

```diff
@@ -60,61 +83,151 @@ namespace $ {
 			return this.products_table().map( row => this.product( row.R_MATERIAL_ID ) )
 		}
 		
-		@ $mol_mem()
-		product( code : string ) {
+		@ $mol_mem_key()
+		product( id : string ) {
 			const next = new $mol_app_inventory_domain_product
-			next.code = $mol_const( code )
-			next.title = ()=> this.product_rows_by_id()[ code ].R_NAME
+			next.id = $mol_const( id )
+			next.code = ()=> this.product_code( id )
```

Comment:
If every time when declared arrow function with parameter, you add space, please add spaces everywhere.

for example (  `param  => { code }`, `() => { code } ` )